File: test/ids.test.ts
import { describe, expect, test } from "bun:test";
import { normalizeId } from "../src/lib/ids.ts";

describe("normalizeId", () => {
  test("adds dashes to 32-char hex string", () => {
    expect(normalizeId("30a61d9c1112802f95fef30d3a601ec5")).toBe(
      "30a61d9c-1112-802f-95fe-f30d3a601ec5",
    );
  });

  test("passes through already-dashed UUID", () => {
    expect(normalizeId("30a61d9c-1112-802f-95fe-f30d3a601ec5")).toBe(
      "30a61d9c-1112-802f-95fe-f30d3a601ec5",
    );
  });

  test("handles uppercase hex", () => {
    expect(normalizeId("30A61D9C1112802F95FEF30D3A601EC5")).toBe(
      "30a61d9c-1112-802f-95fe-f30d3a601ec5",
    );
  });

  test("returns non-UUID strings as-is", () => {
    expect(normalizeId("not-a-uuid")).toBe("not-a-uuid");
    expect(normalizeId("")).toBe("");
    expect(normalizeId("12345")).toBe("12345");
  });

  test("returns strings with non-hex chars as-is", () => {
    expect(normalizeId("30a61d9c1112802f95fef30d3a601ecZ")).toBe(
      "30a61d9c1112802f95fef30d3a601ecZ",
    );
  });
});

File: internal/ids/ids_test.go
package ids

import "testing"

func TestNormalize(t *testing.T) {
	cases := []struct{ in, want string }{
		{"30a61d9c1112802f95fef30d3a601ec5", "30a61d9c-1112-802f-95fe-f30d3a601ec5"},
		{"30a61d9c-1112-802f-95fe-f30d3a601ec5", "30a61d9c-1112-802f-95fe-f30d3a601ec5"},
		{"30A61D9C1112802F95FEF30D3A601EC5", "30a61d9c-1112-802f-95fe-f30d3a601ec5"},
		{"not-a-uuid", "not-a-uuid"},
		{"", ""},
		{"12345", "12345"},
		{"30a61d9c1112802f95fef30d3a601ecZ", "30a61d9c1112802f95fef30d3a601ecZ"},
	}
	for _, c := range cases {
		if got := Normalize(c.in); got != c.want {
			t.Errorf("Normalize(%q) = %q, want %q", c.in, got, c.want)
		}
	}
}
